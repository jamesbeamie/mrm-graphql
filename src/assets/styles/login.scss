/* Importing the font */
@import './variables';
@import '../../utils/images/images';

/* body */

body.home-bg {
  background-image: url($land-background);
  background-repeat: no-repeat;
  background-size: 100%;
  text-align: center;
}

body,
html {
  min-height: 100%;
}

/* Converge Title */

header.home-header>h1 {
  height: 46px;
  color: map-get($color-white, 000);
  font-size: 36px;
  font-weight: 600px;
  line-height: 46px;
  margin: 0;
}

header.home-header {
  margin-top: 11.3%;
  margin-bottom: 2%;
}

/*Converge Introduction*/

#converge-intro {
  height: 62px;
  width: 459px;
  color: map-get($color-white, 000);
  font-size: 24px;
  font-weight: lighter;
  line-height: 31px;
  margin-bottom: 1.5%;
  margin-left: auto;
  margin-right: auto;
}

/* Buttons */

.btn-signin {
  display: inline-block;
  border-radius: 6px 6px 6px 6px;
  outline: none;
}

/* Tablet */

#dark-tablet {
  display: block;
  height: 60%;
  width: 54.6%;
  margin-left: auto;
  margin-right: auto;
  margin-top: 6.2%;
  margin-bottom: 5%;
}

/* Media Queries*/

// large desktops
@media only screen and (min-width: 1441px) {
  header.home-header {
    margin-top: 11.5%;
    margin-bottom: 3%;
  }
  #converge-intro {
    margin-bottom: 2%;
  }
  #dark-tablet {
    margin-top: 7%;
    margin-bottom: 7%;
  }
}

// classes to theme the snack bar
.error-theme {
  background-color: $color-danger;
}

.button-error {
  color: map-get($color-white, 000) !important;
}

// small and medium screens
@media all and (max-width: 767px) {
  #dark-tablet {
    display: none !important;
  }
}
