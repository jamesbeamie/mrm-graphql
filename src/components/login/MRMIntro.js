import React from 'react';

/**
 * The welcome message on login page
 *
 * @returns {JSX}
 */
export default () => (
  <div id="converge-intro">
    Meet the Meeting Room App
    <br />
    that your meeting room app aspires to be.
  </div>
);
