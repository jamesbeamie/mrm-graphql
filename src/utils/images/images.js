
export const analyticsIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fanalytics_icon.svg?alt=media&token=726ebd59-7376-4d08-80e7-c483aacca1c6';
export const appBookingsIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fapp_bookings.svg?alt=media&token=7e577c9b-2f6a-4843-aaa4-c5cf58127860';
export const averageAttendeesChartIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Faverage_attendees_chart.svg?alt=media&token=542b3244-9593-41e6-8d45-91118bb1f5af';
export const averageMeetingCharIcont = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Faverage_meeting_chart.svg?alt=media&token=ddf742cf-5e64-4e4c-aa47-c0cc08fc5990';
export const averageRoomChartIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Faverage_room_chart.svg?alt=media&token=61b659f5-3cba-49fc-9d92-8085d03fcfd7';
export const averageRoomUtilisationIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Faverage_room_utilisation.svg?alt=media&token=5ca16922-08ea-4684-a962-00c4f53d62a6';
export const blueTickIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fblue_tick.svg?alt=media&token=d295fcac-46ba-4808-b670-50c9036c2665';
export const loginButtonIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FButton.png?alt=media&token=fb7692b6-1465-47c5-94f6-c94aff31c275';
export const calendarToday24pxIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fcalendar_today_24px.svg?alt=media&token=fa0f94c4-8916-4eeb-918f-68c9793b5b4d';
export const calendarTodayGrey24pxIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fcalendar_today_grey_24px.svg?alt=media&token=5e341363-e186-4c78-bce9-050ace3d94fe';
export const CancelIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FCancel.svg?alt=media&token=300295b5-0c02-4659-8747-f6d48bd38e34';
export const caretBlueIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fcaret_blue.svg?alt=media&token=61aaa23d-07ff-4449-aa57-312fcf00d965';
export const caretWhiteIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fcaret_white.svg?alt=media&token=9a968e6b-ea82-405b-bc17-114e094abcfc';
export const chartAttendeesIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fchart_attendees.svg?alt=media&token=00331636-9f81-4317-9ae6-d5c636046557';
export const chartBlueIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fchart_blue.svg?alt=media&token=2b0fdfcc-4e52-474b-a761-17f5bb521e4a';
export const chevronIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fchevron.svg?alt=media&token=54c2d83a-0834-44dd-8195-53b5ac74e2ff';
export const convergeLogoIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fconverge_logo.svg?alt=media&token=cd6a1244-91bb-4a1b-8d74-696bd42053ac';
export const darkTabletIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FDark%20Tablet.png?alt=media&token=75ad56c4-f7f1-4322-91ce-b4850fa0eeb3';
export const dateRange24pxIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fdate_range_24px.svg?alt=media&token=a644c232-c9b0-4ca0-b4c6-c75ffa851f0a';
export const deleteIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fdelete.svg?alt=media&token=ac2828f2-841e-48bf-8c51-82ffabe54c58';
export const downArrowIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fdown-arrow.png?alt=media&token=9cd0f129-299e-4b74-8657-f9f52fb570bc';
export const download24pxIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fdownload_24px.svg?alt=media&token=2193b543-2fcd-4951-8687-69cbd475ad89';
export const editIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fedit.svg?alt=media&token=9258a9f7-cdde-4e32-974b-3f82525045cd';
export const feedbackIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ffeedback_icon.svg?alt=media&token=513137e0-af71-4e9f-874d-10d1fa8445b6';
export const filterWhiteIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ffilter_white.svg?alt=media&token=359a633b-94ee-4ce2-80f8-334dee3eb7fd';
export const filterYellowIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ffilter_yellow.svg?alt=media&token=b4013c00-7016-4aed-aaed-ba3d5184288e';
export const greenStarIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fgreen_star.svg?alt=media&token=9a3e6af4-75a0-4f7a-bee3-9ebcdc1ab5cc';
export const greyStarIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fgrey_star.svg?alt=media&token=625713a2-2966-420d-bd10-ff6ac7e3cf6d';
export const groupIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FGroup.svg?alt=media&token=7f06b0bb-46bb-456a-ac35-9eeab8a16362';
export const helpOutline = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911';
export const landingBackground = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FLanding%20Background.svg?alt=media&token=324528d4-e813-4f7a-88f6-b3e2145e9a7b';
export const locationIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Flocation_icon.svg?alt=media&token=ccfa3173-d2a1-442f-8af0-30d53a14be7d';
export const maintenance = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fmaintenance.svg?alt=media&token=33c9db8c-721b-4013-a204-9c3e39109991';
export const more = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fmore.svg?alt=media&token=b7a7a45b-9dc5-49f7-85dc-bc235904a792';
export const notificationsIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fnotifications_icon.svg?alt=media&token=23b8d4c8-1c47-4591-bb55-2a97807675fd';
export const place = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fplace_24px.svg?alt=media&token=41d9a604-9cf6-4cde-984f-e515e33e6392';
export const placeholder = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fplaceholder.svg?alt=media&token=4cc06a44-c922-42a6-9d32-51d9179cabe6';
export const poll = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fpoll.svg?alt=media&token=b0c632bd-3f9d-4cd2-b9dd-60aea1550e46';
export const pollRed = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fpoll_red.svg?alt=media&token=f162d826-43a7-4bcb-aac8-0077ea6a4885';
export const profileIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fprofile.svg?alt=media&token=7be801d0-a155-49ae-97ea-5ae6f606b09f';
export const questionIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fprofile.svg?alt=media&token=7be801d0-a155-49ae-97ea-5ae6f606b09f';
export const responseIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fresponse-icon.svg?alt=media&token=51554781-b9e0-4bf8-96b9-434de0bf2a71';
export const responses = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fresponses.svg?alt=media&token=a6560a83-1e31-4401-83cc-abdc42c246f9';
export const searchIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fsearch_icon.svg?alt=media&token=ef28a273-f3ea-4370-9720-845bae1cd05e';
export const settingsGearsIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fsettings_gears.svg?alt=media&token=f8e21af6-70f7-4c0a-9115-b6013712abbb';
export const settingsIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fsettings_icon.svg?alt=media&token=e103292e-c233-480b-9199-6df0576e5683';
export const threeDotsIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fthree-dots.svg?alt=media&token=9950e1bf-6fc9-4595-8779-78680c3cff88';
export const timeline24Icon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ftimeline_24px.svg?alt=media&token=6f703b51-25c4-4362-b91f-733668bf1bf6';
export const timelineYellowIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ftimeline_yellow_24px.svg?alt=media&token=b1250514-d1ab-45a3-8e16-e33556f71657';
export const timerBlueIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ftimer_blue.svg?alt=media&token=a6e2b74b-986e-4d5e-a6b1-b644387aaec4';
export const warningIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fwarning_icon.svg?alt=media&token=38fa5d2e-595c-4032-a709-9d11c0554b62';
export const welcomeImage = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fwelcome.png?alt=media&token=1665ecda-2592-485d-8f23-b3973dd9905a';
export const walkingIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FWalkingIcon.svg?alt=media&token=04cb5833-46fd-464d-b8c8-6cca7156a2a9';
export const ellipses = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FEllipses.svg?alt=media&token=25a2ecb1-801e-4b9a-bb9d-8399faaee938';
export const roomIcon = 'https://firebasestorage.googleapis.com/v0/b/andelamrm.appspot.com/o/MrmFrontImagesRoom%20Icon.svg?alt=media&token=d2ed6ade-acb5-4f62-a599-ebc752e245c8';
export const resourcesIcom = 'https://firebasestorage.googleapis.com/v0/b/andelamrm.appspot.com/o/MrmFrontImagesResources%20Icon_Inactive.svg?alt=media&token=478d170d-fd35-4ff4-94b7-e7ab88830bf9';
export const structureIcon = 'https://firebasestorage.googleapis.com/v0/b/andelamrm.appspot.com/o/MrmFrontImagesStructure%20Icon.svg?alt=media&token=4c54d430-6b3d-4a76-a3d8-393fb2633f39';
export const deviceIcon = 'https://firebasestorage.googleapis.com/v0/b/andelamrm.appspot.com/o/MrmFrontImagesDevice%20Icon.svg?alt=media&token=0a614b7b-6263-4083-abf7-e7a17ae467c9';
export const defaultUserIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fsilhouette200.png?alt=media&token=e7a6ee92-ef10-411a-814d-95a077585a3e';
export const peopleIcon = 'https://firebasestorage.googleapis.com/v0/b/andelamrm.appspot.com/o/MrmFrontImagesPeople%20Icon.svg?alt=media&token=4decf368-0c30-44ae-af46-e7be23c981fd%27';
export const wingsIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FIllustration_wings.svg?alt=media&token=791562e1-9463-4e7f-9e55-152800bbe5ba';
export const blockIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FIllustration_block.svg?alt=media&token=23a1b058-acc0-42e2-a0c2-b6cd4b655e30';
export const notFoundIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FnotFoundIcon.svg?alt=media&token=7af53714-71a3-4c6f-bdf5-bbca9a170b8e';
export const addIcon = 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FaddIcon.svg?alt=media&token=3bc40a49-1ded-4090-be03-f63f94b50eef';
