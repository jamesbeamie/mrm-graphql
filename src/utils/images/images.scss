$blue-tick : 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fblue_tick.svg?alt=media&token=d295fcac-46ba-4808-b670-50c9036c2665';
$cancel-image: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FCancel.svg?alt=media&token=300295b5-0c02-4659-8747-f6d48bd38e34';
$calendar-today: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fcalendar_today_24px.svg?alt=media&token=fa0f94c4-8916-4eeb-918f-68c9793b5b4d';
$caret-blue: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fcaret_blue.svg?alt=media&token=61aaa23d-07ff-4449-aa57-312fcf00d965';
$caret-white: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fcaret_white.svg?alt=media&token=9a968e6b-ea82-405b-bc17-114e094abcfc';
$date-range: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fdate_range_24px.svg?alt=media&token=a644c232-c9b0-4ca0-b4c6-c75ffa851f0a';
$filter-white: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ffilter_white.svg?alt=media&token=359a633b-94ee-4ce2-80f8-334dee3eb7fd';
$filter-yellow: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ffilter_yellow.svg?alt=media&token=b4013c00-7016-4aed-aaed-ba3d5184288e';
$green-star: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fgreen_star.svg?alt=media&token=9a3e6af4-75a0-4f7a-bee3-9ebcdc1ab5cc';
$land-background: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FLandingBackground.svg?alt=media&token=e8f199b6-608b-4180-881b-eb60e3176c15';
$location-icon: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Flocation_icon.svg?alt=media&token=ccfa3173-d2a1-442f-8af0-30d53a14be7d';
$place-holder: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fplaceholder.svg?alt=media&token=4cc06a44-c922-42a6-9d32-51d9179cabe6';
$response-icon: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fresponse-icon.svg?alt=media&token=51554781-b9e0-4bf8-96b9-434de0bf2a71';
$responses : 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fresponses.svg?alt=media&token=a6560a83-1e31-4401-83cc-abdc42c246f9';
$three-dots: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fthree-dots.svg?alt=media&token=9950e1bf-6fc9-4595-8779-78680c3cff88';
$timeline-yellow: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ftimeline_yellow_24px.svg?alt=media&token=b1250514-d1ab-45a3-8e16-e33556f71657';
$question-icon: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fquestion-icon.svg?alt=media&token=54f8c502-162f-4f6a-b90a-d9883a4df0ff';
$timer-blue: 'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Ftimer_blue.svg?alt=media&token=a6e2b74b-986e-4d5e-a6b1-b644387aaec4';
$chevron-icon:'https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fchevron.svg?alt=media&token=54c2d83a-0834-44dd-8195-53b5ac74e2ff';
$image-placeholder-bg: 'https://i.postimg.cc/VLC3WF9t/3pk-Fb-IT7-rn.gif';
