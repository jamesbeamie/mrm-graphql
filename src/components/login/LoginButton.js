import React from 'react';
import { loginButtonIcon } from '../../utils/images/images';

/**
 * Login Button component
 *
 * @returns {JSX}
 */
const LoginButton = () => {
  // get urls from the .env file
  const { ANDELA_LOGIN_URL, MRM_URL } = process.env || {};

  return (
    <a href={`${ANDELA_LOGIN_URL}=${MRM_URL}`}>
      <input
        type="image"
        src={loginButtonIcon}
        alt="Login With Your Email"
        className="btn-signin"
      />
    </a>
  );
};

export default LoginButton;
