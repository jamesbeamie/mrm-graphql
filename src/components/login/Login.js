import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Snackbar } from 'react-toolbox';
import { withRouter } from 'react-router-dom';

import ROUTES from '../../utils/routes';
import '../../assets/styles/login.scss';
import { darkTabletIcon } from '../../utils/images/images';
import LoginButton from '../login/LoginButton';
import MRMIntro from '../login/MRMIntro';
import { decodeTokenAndGetUserData, getToken } from '../../utils/Cookie';
import Constants from '../../utils/Constants';
import {
  parseQueryString,
  getItemFromLocalStorage,
  saveItemInLocalStorage,
} from '../../utils/Utilities';

// destructor constants to be used
const {
  MRM_TOKEN,
  MESSAGES: { defaultLoginError },
} = Constants;

/**
 * The login component.
 *
 * @extends Component
 *
 * @returns {JSX}
 */
export class Login extends Component {
  static propTypes = {
    location: PropTypes.shape({
      pathname: PropTypes.string.isRequired,
      search: PropTypes.string,
      state: PropTypes.object,
    }).isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }).isRequired,
  };

  state = {
    loginError: null,
  };

  /**
   * Gets the derived state from the props
   *
   * @param {array} props
   *
   * @returns {void}
   */
  static getDerivedStateFromProps = (props) => {
    const { push } = props.history;
    const token = getItemFromLocalStorage(MRM_TOKEN);
    const activeTab = sessionStorage.getItem('activeTopNav') || ROUTES.analytics;
    // this logic has been placed in this lifecycle method to avoid rendering of
    // the login page UI if a user is already loggedin
    // if user is logged in
    if (token) {
      // redirect user to settings page
      push(activeTab);
    }
    // returning null since no state updates
    return null;
  };

  componentDidMount = () => {
    const { push } = this.props.history;

    // apply body styles
    this.applyBodyStyles();

    // parse cookie to see if a user has been succesfully logged in by the Andela login api
    const tokenFromCookies = getToken();
    // decode token to get user data
    const data = decodeTokenAndGetUserData();
    if (data && 'UserInfo' in data) {
      // save token to local storage
      saveItemInLocalStorage(MRM_TOKEN, tokenFromCookies);
      // redirect user to settings page
      push(ROUTES.settings);
      return; // avoid execution of error handling if no errors occur
    }

    // handle login errors from the Andela API if they occur
    const { search = '', state: locationState } = this.props.location;
    const queryParameters = parseQueryString(search) || {};
    if ('error' in queryParameters) {
      this.handleLoginError();
    }
    // handle errors that occur when a user tries to access a protected page
    if (locationState && 'errorMessage' in locationState) {
      this.handleLoginError(locationState.errorMessage);
    }
  };

  componentWillUnmount() {
    // remove home-bg class when unmounting this component to another component
    document.body.classList.remove('home-bg');
  }

  /**
   * Apply body styles on homepage only
   *
   * @returns {void}
   */
  applyBodyStyles = () => {
    const { location } = this.props;
    location.pathname === ROUTES.home && document.body.classList.add('home-bg');
  };

  /**
   * shows login error
   *
   * @returns {void}
   */
  handleLoginError = (loginError = defaultLoginError) => {
    this.setState({ loginError });
  };

  /**
   * It closes the snack bar and redirects to home page
   *
   * @returns {void}
   */
  handleSnackBarClose = () => {
    this.setState({ loginError: null });
    this.props.history.push(ROUTES.home);
  };

  render() {
    const { loginError } = this.state;
    return (
      <Fragment>
        <header className="home-header">
          <h1>CONVERGE</h1>
        </header>
        <MRMIntro />
        <LoginButton />

        <img src={darkTabletIcon} alt="Dark Tablet" id="dark-tablet" />

        {/* display login errors */}
        {loginError && (
          <Snackbar
            type="cancel"
            action="Dismiss"
            label={loginError}
            timeout={Constants.ERROR_TIMEOUT}
            active={Boolean(loginError)}
            onTimeout={this.handleSnackBarClose}
            onClick={this.handleSnackBarClose}
            theme={{ snackbar: 'error-theme', button: 'button-error' }}
          />
        )}
      </Fragment>
    );
  }
}

export default withRouter(Login);
