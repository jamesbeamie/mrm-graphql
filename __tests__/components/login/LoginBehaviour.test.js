/**
 * File tests some behaviour of the LoginComponent
 * Was placed here because mocking files in Login.test.js will make some tests fail
 * And putting these tests at the bottom will raise linting errors
 *
 */
import React from 'react';
import { mount } from 'enzyme';
import { Login } from '../../../src/components/login/Login';
import { profileIcon } from '../../../src/utils/images/images';

// creating mock names to use
const mockFirstName = 'MockFirstName';
const mockLastName = 'MockLastName';

jest.mock('./../../../src/utils/Utilities');
const utilityFUnctions = require('./../../../src/utils/Utilities');

// mock the method that gets token from local storage and make it return a mock token
utilityFUnctions.getItemFromLocalStorage.mockImplementation(() => 'MOCK_TOKEN');

jest.mock('./../../../src/utils/Cookie');
const cookieFunctions = require('../../../src/utils/Cookie');

describe('Login component', () => {
  const push = jest.fn();
  let loginWrapper;
  beforeEach(() => {
    loginWrapper = mount(<Login location={{ pathname: '/' }} history={{ push }} />);
  });
  it('should redirect to another page when a user successfully logs in and his token decoded', () => {
    // implementing the method mock for decoding user data and make it return mock user data
    cookieFunctions.decodeTokenAndGetUserData.mockImplementation(() => ({
      UserInfo: {
        picture: profileIcon,
        first_name: mockFirstName,
        last_name: mockLastName,
      },
    }));

    expect(push).toHaveBeenCalled();
    expect(loginWrapper).toMatchSnapshot();
  });

  it('should redirect to another page when a token is present in localstorage', () => {
    // create a mock token in the local storage
    expect(push).toHaveBeenCalled();
    expect(loginWrapper).toMatchSnapshot();
  });
});
