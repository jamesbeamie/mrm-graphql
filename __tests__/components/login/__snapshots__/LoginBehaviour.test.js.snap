// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should redirect to another page when a token is present in localstorage 1`] = `
<Login
  history={
    Object {
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/analytics",
          ],
          Array [
            "/analytics",
          ],
          Array [
            undefined,
          ],
        ],
      },
    }
  }
  location={
    Object {
      "pathname": "/",
    }
  }
>
  <header
    className="home-header"
  >
    <h1>
      CONVERGE
    </h1>
  </header>
  <Component>
    <div
      id="converge-intro"
    >
      Meet the Meeting Room App
      <br />
      that your meeting room app aspires to be.
    </div>
  </Component>
  <LoginButton>
    <a
      href="undefined=undefined"
    >
      <input
        alt="Login With Your Email"
        className="btn-signin"
        src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FButton.png?alt=media&token=fb7692b6-1465-47c5-94f6-c94aff31c275"
        type="image"
      />
    </a>
  </LoginButton>
  <img
    alt="Dark Tablet"
    id="dark-tablet"
    src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FDark%20Tablet.png?alt=media&token=75ad56c4-f7f1-4322-91ce-b4850fa0eeb3"
  />
</Login>
`;

exports[`Login component should redirect to another page when a user successfully logs in and his token decoded 1`] = `
<Login
  history={
    Object {
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/analytics",
          ],
        ],
      },
    }
  }
  location={
    Object {
      "pathname": "/",
    }
  }
>
  <header
    className="home-header"
  >
    <h1>
      CONVERGE
    </h1>
  </header>
  <Component>
    <div
      id="converge-intro"
    >
      Meet the Meeting Room App
      <br />
      that your meeting room app aspires to be.
    </div>
  </Component>
  <LoginButton>
    <a
      href="undefined=undefined"
    >
      <input
        alt="Login With Your Email"
        className="btn-signin"
        src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FButton.png?alt=media&token=fb7692b6-1465-47c5-94f6-c94aff31c275"
        type="image"
      />
    </a>
  </LoginButton>
  <img
    alt="Dark Tablet"
    id="dark-tablet"
    src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FDark%20Tablet.png?alt=media&token=75ad56c4-f7f1-4322-91ce-b4850fa0eeb3"
  />
</Login>
`;
